src/Buildalyzer/Logging/EventProcessor.cs

```diff
@@ -83,18 +83,22 @@ private void ProjectStarted(object sender, ProjectStartedEventArgs e)
                     _currentResult.Push(result);
                     return;
                 }
+                _currentResult.Push(null);
             }
 
             // Push a null result so the stack is balanced on project finish
-            _currentResult.Push(null);
+            // _currentResult.Push(null);
         }
 
         private void ProjectFinished(object sender, ProjectFinishedEventArgs e)
         {
-            AnalyzerResult result = _currentResult.Pop();
-            if (result != null)
+            if (AnalyzerManager.NormalizePath(e.ProjectFile) == _projectFilePath)

```

Comment:
I suggest extracting it to a function with some meaningful name to explain when this can actually happen :)